### PR TITLE
Critical comments admin comments

### PR DESCRIPTION
**Pull request by jaleszek**
- Critical comments for Admin and Assessor
- Admin comments for Admin

### CODE REVIEW

File: app/assets/javascripts/admin/comments.js.coffee [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ready = ->
-  $('body').on 'submit', '#new_comment', (e)->
+  $('body').on 'submit', '.new_comment', (e)->
```
Comment (matthewford) [inline app/assets/javascripts/admin/comments.js.coffee]: Function arrow (->) must be spaced properly

File: app/assets/javascripts/admin/comments.js.coffee [hunk 2]
```diff
@@ -7,10 +8,10 @@ ready = ->
       data: $(this).serialize(),
       dataType: 'HTML',
       success: (data)->
-        $('#new_comment textarea').val("")
-        $('.comments-container .comment-insert').after(data)
+        that.parents(".comments-container").find("textarea").val("")
+        that.parents(".comments-container").find(".comment-insert").after(data)
 
-  $('body').on 'submit', '.edit_comment', (e)->
+  $('body').on 'submit', '.destroy-comment', (e)->
```
Comment (matthewford) [inline app/assets/javascripts/admin/comments.js.coffee]: Function arrow (->) must be spaced properly

File: app/assets/stylesheets/admin/base.scss [hunk 3]
```diff
@@ -367,10 +367,22 @@ table .ellipsis {
   .icon-flagged {
     background-image: image-url("icon-flag.png");
     background-size: 20px 26px;
+    position: relative;
 
     @include is-retina {
       background-image: image-url("icon-flag@2.png");
     }
+
+    .flag-count{
```
Comment (matthewford) [inline app/assets/stylesheets/admin/base.scss]: Opening curly brace `{` should be preceded by one space

File: app/controllers/assessor/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  def new
+    @comment = form_answer.comments.build(section: "critical")
+    authorize @comment, :create?
+
+    respond_to do |format|
+      format.html { render :new, layout: false }
+    end
+  end
+
+  def create
```
Comment (matthewford) [inline app/controllers/assessor/comments_controller.rb]: Assignment Branch Condition size for create is too high. [17.06/15]

File: app/models/assessor_assignment.rb [hunk 5]
```diff
@@ -21,6 +21,7 @@ class AssessorAssignment < ActiveRecord::Base
 
     validate :submitted_at_immutability
     validate :assessor_existence
+    validate :assessor_assignment_to_category
```
Comment (matthewford) [inline app/models/assessor_assignment.rb]: Inconsistent indentation detected.

File: spec/features/admin/form_answers/comments_management_spec.rb [hunk 6]
```diff
@@ -30,16 +33,49 @@
...
+      it "adds flag to created comment" do
+        first("#admin-comments-heading a").click
+        within admin_comments do
+          populate_comment_form
+          find(".link-flag-comment").click
+          click_button "Comment"
+          expect(page).to have_selector(".comment-flagged", count: 1)
+
+          within ".comment" do
+            find(".link-flag-comment").click #unclick flag
```
Comment (matthewford) [inline spec/features/admin/form_answers/comments_management_spec.rb]: Missing space after #.

**Comment by jaleszek** [comment]
@RuslanHamidullin take a look please in your spare time. Thanks.

File: app/models/comment.rb [hunk 7]
```diff
@@ -6,7 +6,27 @@ class Comment < ActiveRecord::Base
...
+  enum section: {
+    admin: 0,
+    critical: 1
+  }
+
   def author?(subject)
     authorable == subject
   end
+
+  def self.admin
```
Comment (RuslanHamidullin) [inline app/models/comment.rb]: Very minor thing:
I think scopes will be better (but just my opinion - it's matter of taste I think)
```
begin :scopes
  scope :desc ->  { order(created_at: :desc) }
  scope :admin -> { where(section: 0).desc }
  scope :critical -> { where(section: 1).desc }
  scope :flagged -> {  where(flagged: true).desc }
end
```

**Comment by RuslanHamidullin** [comment]
Goodio :+1: :bomb:

**Pull request merged**